packages/cli-repl/src/cli-repl.ts

```diff
@@ -262,22 +265,36 @@ class CliRepl {
     const originalEval = util.promisify(this.repl.eval);
 
     const customEval = async(input, context, filename, callback) => {
+      let result;
+      let err = null;
+
       try {
-        const result = await this.ShellEvaluator.customEval(originalEval, input, context, filename);
-        callback(null, result);
+        result = await this.ShellEvaluator.customEval(originalEval, input, context, filename);
       } catch (err) {
-        callback(err, null);
+        if (isRecoverableError(input)) {
+          return callback(new Recoverable(err));
+        } else {
+          result = err;
+        }
       }
+      callback (null, result)
     };
 
+    // function isRecoverableError(error) {
```

Comment:
Any reason to keep commented function here?

Reply:
i forgot, thanks!